src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketServerConfiguration.java

```diff
@@ -327,5 +341,29 @@ public FormValidation doTestConnection(
         public String getDisplayName() {
             return "Bitbucket Server";
         }
+
+        /**
+         * Overrides the configuration constructor from form data to provide better error handling
+         *
+         * param req request
+         * @param formData json data
+         * @return a new BitbucketSCM instance
+         * @throws FormException if any data is incorrect
+         */
+        @Override
+        public BitbucketServerConfiguration newInstance(@Nullable StaplerRequest req, @Nonnull JSONObject formData) throws FormException {
+            try {
+                return super.newInstance(req, formData);
+            } catch (Error e) {
+                Throwable cause = e.getCause();
+                while (cause != null) {
+                    if (cause instanceof NullPointerException) {
+                        throw new FormException(cause.getMessage(), cause.getMessage());
```

Comment:
I'm not thrilled about this. Could we have our own exception? Maybe we could create our own `requireNonNull` and have it throw a `BitbucketFormException` which has the field and the message?
Here we can then look for our exception and be nice and specific.

Reply:
Would a custom `requireNonNull` live in the server configuration, or elsewhere?

Reply:
I think we would create a `Util` class for that so it can be used by all configuration forms. I imagine something like 
```java
public static <T> T requireNonNull(T value, String field, String message) {
    if (value == null) {
       throw new BitbucketFormException(field, message);
    }
 return value;
}
```